### PR TITLE
Bugfix in dimID index for REICE_MODIS

### PR DESCRIPTION
**Pull request by alejandrobodas**
Minor bug reported by Po-Lun. Wrong index in cosp2_io.f90. This was reported some time ago, but the intel CI test failed (due to issues in the CI test itself). We should be able to fix the CI test if github has updated ubuntu 18.04.

### CODE REVIEW

**Comment by alejandrobodas** [comment]
Hi @RobertPincus , would you mind reviewing these changes? The fortran change is a one-liner, but there are a few changes in the integration test. I have incorporated the changes in #57. The upgrade to Ubuntu 20.04 breaks bit-comparability between gfortran-9 and previous versions, so I had to use a method similar to the one used for ifort in the comparisons against the KGO. Thanks!

File: .github/workflows/continuous_integration.yml [hunk 1]
```diff
@@ -101,7 +101,9 @@ jobs:
         sudo add-apt-repository "deb https://apt.repos.intel.com/oneapi all main"
         sudo apt-get update
         sudo apt-get install intel-hpckit-getting-started intel-oneapi-clck intel-oneapi-common-licensing intel-oneapi-common-vars
-        sudo apt-get install intel-oneapi-dev-utilities  intel-oneapi-dpcpp-cpp-compiler-pro intel-oneapi-ifort intel-oneapi-itac
+        sudo apt-get install intel-oneapi-dev-utilities  intel-oneapi-compiler-dpcpp-cpp-and-cpp-classic intel-oneapi-compiler-fortran intel-oneapi-itac
+        # Ubunu 18.04.3 has an issue that causes ifort to fail invoking ld. packages below avoid the problem
```
Comment (RobertPincus) [inline .github/workflows/continuous_integration.yml:105]: Why not just delete these lines if they are not needed?

File: .github/workflows/continuous_integration.yml [hunk 2]
```diff
@@ -220,12 +229,12 @@ jobs:
...
     - name: Basic against known good output (KGO)
       run: |
-        if [[ -e /opt/intel/oneapi/setvars.sh ]]; then
+        if [[ "${F90}" != 'gfortran' ]]; then
           ATOL=1.0e-20
           RTOL=0.0006
         fi
         cd driver
-        KGO=data/outputs/UKMO/cosp2_output_um.gfortran.kgo.nc
+        KGO=data/outputs/UKMO/cosp2_output_um.gfortran.kgo.$KGO_VERSION.nc
```
Comment (RobertPincus) [inline .github/workflows/continuous_integration.yml:237]: Oooh, I like the versioning of the KGO. Good idea.

File: .github/workflows/continuous_integration.yml [hunk 3]
```diff
@@ -234,18 +243,28 @@ jobs:
...
         else
-          python compare_to_kgo.py ${KGO} ${TST} --atol=${ATOL} --rtol=${RTOL}
+          if [[ "${F90}" = 'gfortran' ]]; then
+            python compare_to_kgo.py ${KGO} ${TST} --atol=${ATOL} --rtol=${RTOL}
+          else
+            ATOL=1.0e-20
+            RTOL=0.0006
+            OUTTST=data/outputs/UKMO/cosp2_output.um_global.out
+            OUTKGO=data/outputs/UKMO/cosp2_output.um_global.gfortran.kgo.$KGO_VERSION.out
+            python compare_to_kgo.py ${KGO} ${TST} --atol=${ATOL} --rtol=${RTOL} \
```
Comment (RobertPincus) [inline .github/workflows/continuous_integration.yml:264]: Rather than repeat the line, maybe move it outside the IF block?

**Review by RobertPincus — APPROVED** [review]
I have a few suggestions for cleanup but the fixes seem fine.

File: .github/workflows/continuous_integration.yml (hunk 1 above)
Comment (alejandrobodas) [reply .github/workflows/continuous_integration.yml]: Done.

File: .github/workflows/continuous_integration.yml (hunk 3 above)
Comment (alejandrobodas) [reply .github/workflows/continuous_integration.yml]: Yes, it was messy.  I'm now using different KGOs for each version, making it easier to adapt in the future.

**Comment by alejandrobodas** [comment]
@RobertPincus , thank you very much for your quick review. I've addressed your comments and all tests have passed, so I'm merging this pull request.

**Pull request merged**